File: backend/app/plugins/simulators/sms/manifest.py
from app.plugins.base import PluginCategory, PluginManifest

manifest = PluginManifest(
    code="sms",
    name="SMS simules",
    name_en="SMS",
    description="Simulation de messages SMS",
    category=PluginCategory.SIMULATOR,
    icon="MessageSquare",
    default_color="gray",
    supported_formats=["text", "image"],
    default_enabled=False,
    coming_soon=False,
    sort_order=7,
    inject_types=["sms"],
    router_module="app.routers.simulated_channels",
    router_prefix="/api/simulated",
)

File: frontend/src/plugins/simulators/sms/manifest.ts
import { lazy } from 'react'
import { MessageSquare } from 'lucide-react'
import type { PluginManifest } from '../../types'

export const manifest: PluginManifest = {
  code: 'sms',
  name: 'plugins.sms.name',
  description: 'plugins.sms.description',
  category: 'simulator',
  icon: MessageSquare,
  defaultColor: 'gray',
  supportedFormats: ['text', 'image'],
  defaultEnabled: false,
  comingSoon: false,
  sortOrder: 7,
  injectTypes: ['sms'],
  playerRoute: '/sms',
  PlayerPage: lazy(() => import('./PlayerPage')),
  EventReceiver: lazy(() => import('./EventReceiver')),
  PreviewPopup: lazy(() => import('./PreviewPopup')),
  configSchema: null,
}
